org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/launch/TestLaunchShortcut.java

```diff
@@ -57,32 +66,30 @@ private void launch(JavaElementResolver resolver) {
             ImmutableList.Builder<TestTarget> targets = ImmutableList.builder();
             targets.addAll(convertTypesToTestTargets(types));
             targets.addAll(convertMethodsToTestTargets(methods));
-            GradleRunConfigurationAttributes runConfigurationAttributes = collectRunConfigurationAttributes(resolver.findFirstContainerProject().get());
+            RunConfiguration runConfigurationAttributes = collectRunConfigurationAttributes(resolver.findFirstContainerProject().get());
             new RunGradleJvmTestLaunchRequestJob(targets.build(), runConfigurationAttributes).schedule();
         } else {
             showNoTestsFoundDialog();
         }
     }
 
     @SuppressWarnings("ConstantConditions")
-    private GradleRunConfigurationAttributes collectRunConfigurationAttributes(IProject project) {
-        FixedRequestAttributes requestAttributes = CorePlugin.projectConfigurationManager().readProjectConfiguration(project).toRequestAttributes(ConversionStrategy.IGNORE_WORKSPACE_SETTINGS);
-
-        // configure the project directory
-        String projectDir = requestAttributes.getProjectDir().getAbsolutePath();
-
-        // configure the advanced options
-        GradleDistribution gradleDistribution = requestAttributes.getGradleDistribution();
-        String javaHome = FileUtils.getAbsolutePath(requestAttributes.getJavaHome()).orNull();
-        List<String> jvmArguments = requestAttributes.getJvmArguments();
-        List<String> arguments = requestAttributes.getArguments();
-
-        // have execution view and console view enabled by default
-        boolean showExecutionView = true;
-        boolean showConsoleView = true;
-
-        return GradleRunConfigurationAttributes.with(ImmutableList.<String>of(), projectDir, gradleDistribution,
-                javaHome, jvmArguments, arguments, showExecutionView, showConsoleView, true);
+    private RunConfiguration collectRunConfigurationAttributes(IProject project) {
+        ProjectConfiguration projectConfig = CorePlugin.configurationManager().loadProjectConfiguration(project);
+        BuildConfiguration buildConfig = projectConfig.getBuildConfiguration();
+        GradleRunConfigurationAttributes attributes = new GradleRunConfigurationAttributes(Collections.<String>emptyList(),
+                buildConfig.getRootProjectDirectory().getAbsolutePath(),
+                GradleDistributionSerializer.INSTANCE.serializeToString(buildConfig.getGradleDistribution()),
+                null,
+                Collections.<String>emptyList(),
+                Collections.<String>emptyList(),
+                true,
+                true,
+                buildConfig.isOverrideWorkspaceSettings(),
+                buildConfig.isOfflineMode(),
+                buildConfig.isBuildScansEnabled());
+        ILaunchConfiguration launchConfig = CorePlugin.gradleLaunchConfigurationManager().getOrCreateRunConfiguration(attributes, SaveStrategy.DONT_PERSIST);
```

Comment:
This doesn't make sense, we shouldn't create an `ILaunch` if we don't need it. We should just have a `configurationManager().createRunconfiguration()` mehod and the launch configuration manager should work with that, not the other way around.